packages/shared/nodes/GraftNode.ts

```diff
@@ -1,39 +1,83 @@
-import { $applyNodeReplacement, EditorConfig, NodeKey } from "lexical";
-import { Attributes, SerializedUsfmElementNode, UsfmElementNode } from "./UsfmElementNode";
+import {
+  $applyNodeReplacement,
+  DOMExportOutput,
+  EditorConfig,
+  LexicalEditor,
+  NodeKey,
+  isHTMLElement,
+} from "lexical";
+import {
+  Attributes,
+  NodeProps,
+  SerializedUsfmElementNode,
+  UsfmElementNode,
+} from "./UsfmElementNode";
 import { addClassNamesToElement } from "@lexical/utils";
 
+const DEFAULT_TAG = "span";
+
 export type SerializedGraftNode = SerializedUsfmElementNode;
 
 export class GraftNode extends UsfmElementNode {
-  constructor(attributes: Attributes, data: unknown, tag?: string, key?: NodeKey) {
-    super(attributes, data, tag, key);
+  constructor(attributes: Attributes = {}, props?: NodeProps, tag?: string, key?: NodeKey) {
+    super(attributes, props, tag, key);
   }
 
   static getType(): string {
     return "graft";
   }
 
   static clone(node: GraftNode): GraftNode {
-    return new GraftNode(node.__attributes, node.__data, node.__tag, node.__key);
+    return new GraftNode(node.__attributes, node.__props, node.__tag, node.__key);
   }
 
   isInline(): boolean {
-    return true;
+    return this.getProps()?.isInline ?? false;
   }
 
   createDOM(config: EditorConfig): HTMLElement {
-    const element = document.createElement(this.getTag() || "span");
+    const element = document.createElement(this.getTag() ?? DEFAULT_TAG);
     const attributes = this.getAttributes() ?? {};
     Object.keys(attributes).forEach((attKey) => {
       element.setAttribute(attKey, attributes[attKey]);
     });
     addClassNamesToElement(element, config.theme.sectionmark);
+    // Setting contenteditable to false is the trick Lexical uses for decorator nodes so you can place the cursor after a span that contains editable content.
+    // element.setAttribute("contenteditable", "false");
```

Comment:
I'm pretty sure this doesn't work on element nodes so you can remove both these comments.